toolchain/check/cpp_type_mapping.cpp

```diff
@@ -26,57 +26,62 @@ namespace Carbon::Check {
 
 // Find the bit width of an integer literal.
 // The default bit width is 32. If the literal's bit width is greater than 32,
-// the bit width is increased to 64. Negative literals are assigned a bit width
-// of 64.
+// the bit width is increased to 64.
 static auto FindIntLiteralBitWidth(Context& context, SemIR::InstId arg_id,
                                    bool is_signed) -> IntId {
-  auto bit_width_id = IntId::MakeRaw(32);
-  auto arg = context.insts().TryGetAs<SemIR::IntValue>(arg_id);
-  if (arg) {
-    auto arg_val = context.ints().Get(arg->int_id);
+  auto arg_const_id = context.constant_values().Get(arg_id);
+  if (!arg_const_id.is_constant() ||
+      arg_const_id == SemIR::ErrorInst::ConstantId ||
+      arg_const_id.is_symbolic()) {
+    // TODO: Add tests for these cases.
+    return IntId::None;
+  }
+  auto arg = context.insts().GetAs<SemIR::IntValue>(
+      context.constant_values().GetInstId(arg_const_id));
+  auto arg_val = context.ints().Get(arg.int_id);
+  unsigned arg_non_sign_bits = arg_val.getSignificantBits() - 1;
 
-    auto width = bit_width_id.has_value()
-                     ? context.ints().Get(bit_width_id).getZExtValue()
-                     : arg_val.getBitWidth();
+  auto bit_width_id = IntId::MakeRaw(32);
+  auto width = context.ints().Get(bit_width_id).getZExtValue();
 
-    unsigned arg_non_sign_bits = arg_val.getSignificantBits() - 1;
-    return (arg_non_sign_bits + is_signed <= width) ? bit_width_id
-                                                    : IntId::MakeRaw(64);
-  }
-  return IntId::MakeRaw(64);
+  // TODO: What if the literal is larger than 64 bits? Currently an error is
+  // reported that the int value is too large for type `i64`. Maybe try to fit
+  // in i128/i256? Try unsigned?
+  return (arg_non_sign_bits + is_signed <= width) ? bit_width_id
+                                                  : IntId::MakeRaw(64);
```

Comment:
This can be simplified a bit:
```suggestion
  // TODO: What if the literal is larger than 64 bits? Currently an error is
  // reported that the int value is too large for type `i64`. Maybe try to fit
  // in i128/i256? Try unsigned?
  return (arg_non_sign_bits + is_signed <= 32) ? IntId::MakeRaw(32)
                                               : IntId::MakeRaw(64);
```

Reply:
Done.